src/olympia/scanners/tests/test_admin.py

```diff
@@ -601,3 +609,214 @@ def test_get_fields_for_non_admins(self):
         request.user = user
         assert 'definition' not in self.admin.get_fields(request=request)
         assert 'formatted_definition' in self.admin.get_fields(request=request)
+
+
+class TestScannerQueryRuleAdmin(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.user = user_factory()
+        self.grant_permission(self.user, 'Admin:ScannersQuery')
+        self.client.login(email=self.user.email)
+        self.list_url = reverse('admin:scanners_scannerqueryrule_changelist')
+
+    def test_list_view(self):
+        ScannerQueryRule.objects.create(name='bar', scanner=YARA)
+        response = self.client.get(self.list_url)
+        assert response.status_code == 200
+
+    def test_list_view_is_restricted(self):
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Curation')
+        self.client.login(email=user.email)
+        response = self.client.get(self.list_url)
+        assert response.status_code == 403
+
+    def test_change_view_contains_link_to_results(self):
+        rule = ScannerQueryRule.objects.create(name='bar', scanner=YARA)
+        result = ScannerQueryResult(scanner=YARA)
+        result.add_yara_result(rule=rule.name)
+        result.save()
+        ScannerQueryResult.objects.create(scanner=YARA)  # Doesn't match
+        url = reverse(
+            'admin:scanners_scannerqueryrule_change', args=(rule.pk,))
+        response = self.client.get(url)
+        assert response.status_code == 200
+        doc = pq(response.content)
+        link = doc('.field-matched_results_link a')
+        assert link
+        results_list_url = reverse(
+            'admin:scanners_scannerqueryresult_changelist')
+        expected_href = (
+            f'{results_list_url}?matched_rules__id__exact={rule.pk}'
+            f'&has_version=all&state=all&scanner={rule.scanner}'
+        )
+        assert link.attr('href') == expected_href
+        assert link.text() == '1'
```

Comment:
Why is the text `1`?

Reply:
See explanation above. I can add/clarify comments in both places if that helps.